src/test/shell/integration/runfiles_test.sh

```diff
@@ -800,4 +800,117 @@ EOF
     //pkg:gen >&$TEST_log || fail "build failed"
 }
 
+function test_runfile_manifest_only_change_not_stale() {
+  if "$is_windows"; then
+    # Can't run shell scripts directly.
+    return
+  fi
+
+  mkdir -p pkg
+  cat > pkg/constants.bzl <<'EOF'
+SYMLINK_PATH = "old_name"
+EOF
+  cat > pkg/defs.bzl <<'EOF'
+load(":constants.bzl", "SYMLINK_PATH")
+
+def _my_script_impl(ctx):
+    out = ctx.actions.declare_file(ctx.label.name)
+    print(SYMLINK_PATH)
+    ctx.actions.write(
+        out,
+        """#!/usr/bin/env sh
+set -eu
+[ -f {} ] || exit 1
```

Comment:
For my understanding: does this script actually need to do anything for the test to work? Why isn't failure to execute sufficient to know that the runfiles tree was not updated?

Reply:
It does, the test executable itself has a stable runfiles path, but the symlink path differs between runs. If we didn't check for file existence, we wouldn't verify that the runfiles tree has been updated.